### PR TITLE
local e2e demo with contract error propagation updates

### PR DESCRIPTION
**Pull request by whalelephant**
- contracts/* - allow the return of `VerifyResult` so that caller can handle appropriately but also easier to show error 
- avida-ts/docker - docker setup for local testnets with hermes relayer
- avida-ts - e2e demo using cosmes to interact with neutron and sd-jwt-js to issue / create presentation for `avida-example` contract

### CODE REVIEW

File: avida-ts/docker/docker-compose.local.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    volumes:
+      - ./scripts/:/tmp
+    entrypoint: /tmp/cheqd-genesis.sh
+    env_file:
+      - ../local.env
+
+  cheqd-resource-uploader:
+    image: ghcr.io/nymlab/cheqd-node:v2.0.1-arm64
+    ports:
+      - target: 26656
```
Comment (lampkin-diet) [inline avida-ts/docker/docker-compose.local.yaml]: I think we don't need to expose ports here cause we do not run the node inside the container.

File: avida-ts/docker/docker-compose.local.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  neutron-deployer:
+    image: ghrc.io/nymlab/neutron-standalone:v3.0.5-arm64
+    volumes:
+      - ./../../artifacts/:/tmp/artifacts/
+      - ./scripts/:/tmp/scripts/
+      - ../contract.env:/tmp/contract.env
+    depends_on:
+      - neutron-node
+    ports:
+      - target: 26656
```
Comment (lampkin-diet) [inline avida-ts/docker/docker-compose.local.yaml]: I assume here the same thing with exposing ports

File: avida-ts/package.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  "name": "avida-ts",
+  "private": true,
+  "version": "1.0.0",
+  "description": "",
+  "scripts": {
+    "ci:lint": "pnpm lint",
+    "ci:check": "pnpm --recursive run check",
+    "setup": "pnpm run reset && pnpm install",
+    "reset": "pnpm dlx rimraf --glob ./**/node_modules",
+    "build": "NODE_ENV=production pnpm run --recursive --stream build",
```
Comment (lampkin-diet) [inline avida-ts/package.json]: Usually, if there are some jobs like `build:txutils` or others it's assumed that `build` action combines all the things together, like:
```
"build": "pnpm run build:txutils && pnpm run build:types && ..."
```
I think it worth to change `build` job into something like `build:production` or other thing which points to exact action.

File: avida-ts/packages/avida-tx-utils/src/sdjwt_utils.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+    return async (data: any) => {
+      const encoder = new TextEncoder();
+      const signature = await sign(null, encoder.encode(data), privateKey);
+      return Buffer.from(signature).toString("base64url");
+    };
+  },
+};
+
+export async function get_sdjwt(privatePEM: string): Promise<SDJwtVcInstance> {
```
Comment (lampkin-diet) [inline avida-ts/packages/avida-tx-utils/src/sdjwt_utils.ts]: let's keep the same style, camelCase for functions.

File: avida-ts/packages/avida-tx-utils/src/upload_and_inst.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    wasmByteCode,
+    sender: deployer.address,
+  });
+
+  const storeTx: UnsignedTx = {
+    msgs: [storeMsg],
+    memo,
+  };
+
+  let fee = await deployer.estimateFee(storeTx);
```
Comment (lampkin-diet) [inline avida-ts/packages/avida-tx-utils/src/upload_and_inst.ts]: Are there any possible errors here could happen? Internet issues or others.
I think it worth at least to use `try-catch` construction.

File: avida-ts/reset.d.ts [hunk 6]
```diff
@@ -0,0 +1,2 @@
+// @see https://www.totaltypescript.com/ts-reset -- do not add any other lines of code to this file
```
Comment (lampkin-diet) [inline avida-ts/reset.d.ts:1]: As I remember, `.d.ts` files are results of generating from `.ts` files and they should not be placed in the repo cause build artifacts. Also, is it used somewhere?

File: avida-ts/scripts/gen-contract-types.mjs [hunk 7]
```diff
@@ -0,0 +1,75 @@
+// @ts-check
```
Comment (lampkin-diet) [inline avida-ts/scripts/gen-contract-types.mjs:1]: The same issue here. Is it generated file and should it be placed in the repo?

**Review by lampkin-diet — COMMENTED** [review]
Hey! Great job and typescript skills 👍 
I just raised some minor things. Let's chat if you have some questions

File: avida-ts/README.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
+# Typescript library for AVIDA
```
Comment (whalelephant) [inline avida-ts/README.md:1]: Add building contract and schemas stages to doc

File: avida-ts/reset.d.ts (hunk 6 above)
Comment (whalelephant) [reply avida-ts/reset.d.ts:1]: This is used in [reset](https://www.totaltypescript.com/ts-reset-article), admittedly not used too much in this code.

File: avida-ts/scripts/gen-contract-types.mjs (hunk 7 above)
Comment (whalelephant) [reply avida-ts/scripts/gen-contract-types.mjs:1]: actually I am struggling with @ts-check for the avida-common-types at the moment in `scripts/local-e2e-demo.mjs`

File: avida-ts/packages/avida-tx-utils/src/upload_and_inst.ts (hunk 5 above)
Comment (whalelephant) [reply avida-ts/packages/avida-tx-utils/src/upload_and_inst.ts]: fixed in https://github.com/nymlab/AVIDA/pull/6/commits/7ce3c1886296d88d19edcfd650036b81c8dcdf75

File: avida-ts/packages/avida-tx-utils/src/sdjwt_utils.ts (hunk 4 above)
Comment (whalelephant) [reply avida-ts/packages/avida-tx-utils/src/sdjwt_utils.ts]: fixed in https://github.com/nymlab/AVIDA/pull/6/commits/7ce3c1886296d88d19edcfd650036b81c8dcdf75

File: avida-ts/package.json (hunk 3 above)
Comment (whalelephant) [reply avida-ts/package.json]: fixedin https://github.com/nymlab/AVIDA/pull/6/commits/7ce3c1886296d88d19edcfd650036b81c8dcdf75

File: avida-ts/docker/docker-compose.local.yaml (hunk 1 above)
Comment (whalelephant) [reply avida-ts/docker/docker-compose.local.yaml]: updated in https://github.com/nymlab/AVIDA/pull/6/commits/7ce3c1886296d88d19edcfd650036b81c8dcdf75

File: avida-ts/docker/docker-compose.local.yaml (hunk 2 above)
Comment (whalelephant) [reply avida-ts/docker/docker-compose.local.yaml]: updated in https://github.com/nymlab/AVIDA/pull/6/commits/7ce3c1886296d88d19edcfd650036b81c8dcdf75

File: avida-ts/scripts/gen-contract-types.mjs (hunk 7 above)
Comment (whalelephant) [reply avida-ts/scripts/gen-contract-types.mjs:1]: checks work in https://github.com/nymlab/AVIDA/pull/6/commits/7ce3c1886296d88d19edcfd650036b81c8dcdf75

File: avida-ts/README.md (hunk 8 above)
Comment (whalelephant) [reply avida-ts/README.md:1]: added doc in https://github.com/nymlab/AVIDA/pull/6/commits/7ce3c1886296d88d19edcfd650036b81c8dcdf75

**Review by lampkin-diet — APPROVED** [review]

**Pull request merged**